Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41068&attempt_id=67a607a3-253e-40ed-ac06-8a48e99e98fa)

Attempted: solc, dotnetPackages.Dafny, seth, dotnetPackages.Boogie, boogie, z3, souper, fstar, cuter, dafny, altcoins.seth

The following builds were skipped because they don't evaluate on aarch64-linux: haskellPackages.sbvPlugin, haskellPackages.z3, haskellPackages.copilot-sbv, cryptol, haskellPackages.ntha, haskellPackages.linearEqSolver, haskellPackages.cryptol, tlaps, isabelle, haskellPackages.sbv, haskellPackages.ez3

<details><summary>Partial log (click to expand)</summary><p>

```
make[2]: Leaving directory '/build/mono-5.8.0.108/runtime'
make[1]: *** [Makefile:549: all-recursive] Error 1
make[1]: Leaving directory '/build/mono-5.8.0.108'
make: *** [Makefile:479: all] Error 2
builder for '/nix/store/5as7c4z3r4h9xjwl96r9jqcbs5fn1yzx-mono-5.8.0.108.drv' failed with exit code 2
cannot build derivation '/nix/store/jh587f41615yqg0zd1k69zvx9r8bdy2j-dotnetbuildhelpers.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lic42f6n9s8x0r5rj7yay23amckg1664-NUnit.Runners-2.6.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/3j3ky26cwig40i3v1g1lz2590bnqrnjf-Boogie-unstable-2017-01-03.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/vlsy07nhc0m375dh8gcgz32j2bh961x4-Dafny-2.1.0.drv': 3 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/3j3ky26cwig40i3v1g1lz2590bnqrnjf-Boogie-unstable-2017-01-03.drv', '/nix/store/kap02c2ky9qsz5mnd6dd4zgivbgc3rzm-fstar-0.9.6.0.drv', '/nix/store/vlsy07nhc0m375dh8gcgz32j2bh961x4-Dafny-2.1.0.drv' failed
```
</p></details>

--- a/pkgs/applications/science/logic/z3/default.nix
+++ b/pkgs/applications/science/logic/z3/default.nix
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   name = "z3-${version}";
-  version = "4.6.0";
+  version = "4.7.1";
 
   src = fetchFromGitHub {
     owner  = "Z3Prover";
     repo   = "z3";
-    rev    = "b0aaa4c6d7a739eb5e8e56a73e0486df46483222";
-    sha256 = "1cgwlmjdbf4rsv2rriqi2sdpz9qxihxrcpm6a4s37ijy437xg78l";
+    rev    = "3b1b82bef05a1b5fd69ece79c80a95fb6d72a990";
+    sha256 = "1s850r6qifwl83zzgvrb5l0jigvmymzpv18ph71hg2bcpk7kjw3d";
   };
 
   buildInputs = [ python fixDarwinDylibNames ];

